101-sql-logical-server/azuredeploy.json

```diff
@@ -56,14 +63,17 @@
        "serverResourceGroupName": "[resourceGroup().name]",
        "subscriptionId": "[subscription().subscriptionId]",
        "uniqueStorage": "[uniqueString(variables('subscriptionId'), variables('serverResourceGroupName'), parameters('location'))]",
-       "storageName": "[tolower(concat('sqlva', variables('uniqueStorage')))]"
+       "storageName": "[tolower(concat('sqlva', variables('uniqueStorage')))]",
+       "uniqueRoleGuid": "[guid(resourceId('Microsoft.Sql/servers', parameters('serverName')))]",
```

Comment:
this should be based on your scope, role and prinicipal... you won't be able to incorporate principal since it's an MSI, but you can fix the scope and role, e.g.

guid(variables(resourceId('Microsoft.Storage/storageAccounts, variables('storageName'), variables('storageBlobContributor'))

Reply:
I can set the guid to be :
 guid(variables(resourceId('Microsoft.Storage/storageAccounts, variables('storageName'), variables('storageBlobContributor'), resourceId('Microsoft.Sql/servers', parameters('serverName'))) 
???

Reply:
I can't use only the storage name and the storageBlobContributor variable.
My storage name is created form - subscription, resource group and location.
If I run this template twice for different servers the last role id will override the first .
So, because of this, I need to add the server name when I create the guid.

Reply:
sounds good